File: src/leetcode/2026/02/Solution594.java
import java.util.HashMap;
import java.util.Map;

public class Solution594 {
    public int findLHS(int[] nums) {
        Map<Integer, Integer> frequencyMap = new HashMap<>();
        for (int num : nums) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }

        int maxLength = 0;
        for (int key : frequencyMap.keySet()) {
            if (frequencyMap.containsKey(key + 1)) {
                int currentLength = frequencyMap.get(key) + frequencyMap.get(key + 1);
                maxLength = Math.max(maxLength, currentLength);
            }
        }

        return maxLength;
    }

    public static void main(String[] args) {
        Solution594 solution = new Solution594();
        int[] nums = { 1, 3, 2, 2, 5, 2, 3, 7 };
        int result = solution.findLHS(nums);
        System.out.println(result); // Should print 5
    }
}

File: src/leetcode/2026/02/Solution594.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution594 {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int> frequencyMap;
        for (int num : nums) {
            frequencyMap[num]++;
        }

        int maxLength = 0;
        for (const auto& pair : frequencyMap) {
            int key = pair.first;
            int value = pair.second;
            if (frequencyMap.find(key + 1) != frequencyMap.end()) {
                int currentLength = value + frequencyMap[key + 1];
                maxLength = max(maxLength, currentLength);
            }
        }

        return maxLength;
    }
};

int main() {
    Solution594 solution;
    vector<int> nums = {1, 3, 2, 2, 5, 2, 3, 7};
    int result = solution.findLHS(nums);
    // Expected output: 5 (the longest harmonious subsequence is [3,2,2,2,3])
    cout << "Length of Longest Harmonious Subsequence: " << result << endl;
    return 0;
}